### PR TITLE
Improve scanner UI

### PR DESCRIPTION
**Pull request by AllForNothing**
1.Refactor report bar for successful status
2.Other modifications
Signed-off-by: sshijun <sshijun@vmware.com>

### CODE REVIEW

File: src/portal/lib/src/tag/tag.component.ts [hunk 1]
```diff
@@ -763,15 +763,27 @@ export class TagComponent implements OnInit, AfterViewInit {
...
+          } else {
+            this.scanBtnState = ClrLoadingState.ERROR;
           }
-          this.scanBtnState = ClrLoadingState.SUCCESS;
         }, error => {
           this.scanBtnState = ClrLoadingState.ERROR;
         });
   }
+  getScannerMetadata(uuid: string) {
+    this.http.get(`/api/scanners/${uuid}/metadata`)
```
Comment (jwangyangls) [inline src/portal/lib/src/tag/tag.component.ts]: This should appear in the service file

File: src/portal/lib/src/vulnerability-scanning/histogram-chart/histogram-chart.component.ts [hunk 2]
```diff
@@ -127,6 +127,6 @@ export class HistogramChartComponent implements OnInit, AfterViewInit, DoCheck {
       });
     }
     this.max = count;
-    this.scale = Math.ceil(count / 4);
+    this.scale = Math.ceil(count / 40) * 10;
```
Comment (jwangyangls) [inline src/portal/lib/src/vulnerability-scanning/histogram-chart/histogram-chart.component.ts]: Please confirm the change of logic here

File: src/portal/lib/src/vulnerability-scanning/result-grid.component.ts [hunk 3]
```diff
@@ -39,11 +45,43 @@ export class ResultGridComponent implements OnInit {
...
+                    this.getScannerMetadata(response.uuid);
+                } else {
+                    this.scanBtnState = ClrLoadingState.ERROR;
+                }
+            }, error => {
+                this.scanBtnState = ClrLoadingState.ERROR;
+            });
+    }
+    getScannerMetadata(uuid: string) {
+        this.http.get(`/api/scanners/${uuid}/metadata`)
```
Comment (jwangyangls) [inline src/portal/lib/src/vulnerability-scanning/result-grid.component.ts]: abort

File: src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.scss [hunk 4]
```diff
@@ -220,3 +223,54 @@ hr {
 .width-150 {
     width: 150px;
 }
+.circle-block {
+    color: #575757;
+    display: flex;
+    align-items: center;
+    :first-child {
```
Comment (zhoumeina) [inline src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.scss]: Forgot &?

File: src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.ts [hunk 5]
```diff
@@ -72,7 +82,21 @@ export class ResultTipHistogramComponent implements OnInit {
 
     return 0;
   }
-
+  duration(): string {
+    if (this.vulnerabilitySummary && this.vulnerabilitySummary.duration) {
+      let str = '';
+      let min = Math.floor(this.vulnerabilitySummary.duration / MIN);
```
Comment (jwangyangls) [inline src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.ts]: should be const

File: src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.ts [hunk 6]
```diff
@@ -72,7 +82,21 @@ export class ResultTipHistogramComponent implements OnInit {
...
+      if (min) {
+        str += min + MIN_STR;
+      }
+      let sec = this.vulnerabilitySummary.duration % MIN;
```
Comment (jwangyangls) [inline src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.ts]: abort

File: src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.ts [hunk 7]
```diff
@@ -139,6 +163,29 @@ export class ResultTipHistogramComponent implements OnInit {
     return this.total === 0;
   }
 
+  getClass(): string {
+    if (this.vulnerabilitySummary && this.vulnerabilitySummary.severity) {
+      if (this.isCritical) {
+        return 'level-critical';
```
Comment (jwangyangls) [inline src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.ts]: hard code

File: src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.scss (hunk 4 above)
Comment (AllForNothing) [reply src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.scss]: Just confirmed, it's right.

File: src/portal/lib/src/vulnerability-scanning/histogram-chart/histogram-chart.component.ts (hunk 2 above)
Comment (AllForNothing) [reply src/portal/lib/src/vulnerability-scanning/histogram-chart/histogram-chart.component.ts]: Confirmed, it's right

File: src/portal/lib/src/tag/tag.component.ts (hunk 1 above)
Comment (zhoumeina) [reply src/portal/lib/src/tag/tag.component.ts]: +1

File: src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.scss (hunk 4 above)
Comment (zhoumeina) [reply src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.scss]: .circle-block {
    color: #575757;
    display: flex;
    align-items: center;
    >div:first-child {
        display: inline-block;
        border-radius: 50%;
        height: 20px;
        width: 20px;
        line-height: 20px;
        font-size: 14px;
        text-align: center;
    }
}

File: src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.scss [hunk 8]
```diff
@@ -220,3 +223,54 @@ hr {
...
+        border-radius: 50%;
+        height: 20px;
+        width: 20px;
+        line-height: 20px;
+        font-size: 14px;
+        text-align: center;
+    }
+}
+
+.level-critical {
```
Comment (zhoumeina) [inline src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.scss]: border should be set in a single css style. Then you can use it in the template.

File: src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.ts (hunk 5 above)
Comment (AllForNothing) [reply src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.ts]: fixed

File: src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.ts (hunk 6 above)
Comment (AllForNothing) [reply src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.ts]: fixed

File: src/portal/lib/src/tag/tag.component.ts (hunk 1 above)
Comment (AllForNothing) [reply src/portal/lib/src/tag/tag.component.ts]: fixed

File: src/portal/lib/src/vulnerability-scanning/result-grid.component.ts (hunk 3 above)
Comment (AllForNothing) [reply src/portal/lib/src/vulnerability-scanning/result-grid.component.ts]: fixed

File: src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.scss (hunk 8 above)
Comment (AllForNothing) [reply src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.scss]: fixed

File: src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.scss (hunk 4 above)
Comment (AllForNothing) [reply src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.scss]: Modified

**Pull request merged**